Runtime error when using dolt_commit_diff_$tablename

I've gotten this runtime error now multiple times when diffing, it seems like it's maybe related to table not existing (or having some radical changes such as table being removed and added again etc.) between from_revision or to_revision. It's pretty bad since the panic will drop the whole server, instead some error would be nice.

I think I've gotten same error if both from_revision and to_revision hash are the same, also having uncommitted changes on the table might be a factor. Hard to say, haven't been able to get solid repro on this.

Happens at least on versions 0.75.9 and 0.75.10.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x8 pc=0x105fbdad8]

goroutine 2099 [running]:
github.com/dolthub/dolt/go/libraries/doltcore/doltdb.(*Table).NodeStore(...)
	github.com/dolthub/dolt/go/libraries/doltcore/doltdb/table.go:129
github.com/dolthub/dolt/go/libraries/doltcore/sqle/dtables.newProllyDiffIter(_, {_, _, {_, _}, {_, _}, _, _, {_, ...}, ...}, ...)
	github.com/dolthub/dolt/go/libraries/doltcore/sqle/dtables/diff_iter.go:275 +0x3e8
github.com/dolthub/dolt/go/libraries/doltcore/sqle/dtables.DiffPartition.GetRowIter({0x0, 0x0, {0x140011bbde0, 0x20}, {0x140011bbc80, 0x20}, 0x14001952048, 0x14001952060, {0x106d74a60, 0x14001286a20}, ...}, ...)
	github.com/dolthub/dolt/go/libraries/doltcore/sqle/dtables/diff_table.go:640 +0xf0
github.com/dolthub/dolt/go/libraries/doltcore/sqle/dtables.(*CommitDiffTable).PartitionRows(0x1400279e9b8?, 0x10560d048?, {0x106d44180?, 0x14001286ea0?})
	github.com/dolthub/dolt/go/libraries/doltcore/sqle/dtables/commit_diff_table.go:307 +0xb0
github.com/dolthub/go-mysql-server/sql/plan.(*ProcessTable).PartitionRows(0x14001339d70, 0x104e7db74?, {0x106d44180, 0x14001286ea0})
	github.com/dolthub/go-mysql-server@v0.14.1-0.20230329021806-7d5325285657/sql/plan/process.go:285 +0x38
github.com/dolthub/go-mysql-server/sql.(*TableRowIter).Next(0x14001809c20, 0x14000d4d040)
	github.com/dolthub/go-mysql-server@v0.14.1-0.20230329021806-7d5325285657/sql/table_iter.go:60 +0x14c
github.com/dolthub/go-mysql-server/sql/plan.(*projectIter).Next(0x1400127c0f0, 0x104e3e3d0?)
	github.com/dolthub/go-mysql-server@v0.14.1-0.20230329021806-7d5325285657/sql/plan/project.go:157 +0x34
github.com/dolthub/go-mysql-server/sql/plan.transactionCommittingIter.Next(...)
	github.com/dolthub/go-mysql-server@v0.14.1-0.20230329021806-7d5325285657/sql/plan/transaction_committing_iter.go:118
github.com/dolthub/go-mysql-server/sql/plan.(*trackedRowIter).Next(0x14001809c70, 0x104e485d0?)
	github.com/dolthub/go-mysql-server@v0.14.1-0.20230329021806-7d5325285657/sql/plan/process.go:387 +0x30
github.com/dolthub/go-mysql-server/server.(*Handler).doQuery.func2()
	github.com/dolthub/go-mysql-server@v0.14.1-0.20230329021806-7d5325285657/server/handler.go:413 +0x1fc
golang.org/x/sync/errgroup.(*Group).Go.func1()
	golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75 +0x5c
created by golang.org/x/sync/errgroup.(*Group).Go
	golang.org/x/sync@v0.1.0/errgroup/errgroup.go:72 +0x9c
```

--- a/go/libraries/doltcore/sqle/dtables/diff_iter.go
+++ b/go/libraries/doltcore/sqle/dtables/diff_iter.go
@@ -233,7 +233,7 @@ var _ sql.RowIter = prollyDiffIter{}
 // than |targetFromSchema| or |targetToSchema|. We convert the rows from the
 // schema of |from| to |targetFromSchema| and the schema of |to| to
 // |targetToSchema|. See the tablediff_prolly package.
-func newProllyDiffIter(ctx *sql.Context, dp DiffPartition, ddb *doltdb.DoltDB, targetFromSchema, targetToSchema schema.Schema) (prollyDiffIter, error) {
+func newProllyDiffIter(ctx *sql.Context, dp DiffPartition, targetFromSchema, targetToSchema schema.Schema) (prollyDiffIter, error) {
 	fromCm := commitInfo2{
 		name: dp.fromName,
 		ts:   (*time.Time)(dp.fromDate),

--- a/go/libraries/doltcore/sqle/dtables/diff_table.go
+++ b/go/libraries/doltcore/sqle/dtables/diff_table.go
@@ -663,7 +663,7 @@ func (dp *DiffPartition) isDiffablePartition(ctx *sql.Context) (bool, error) {
 	if err != nil {
 		return false, err
 	}
-	
+
 	toSch, err := dp.to.GetSchema(ctx)
 	if err != nil {
 		return false, err

--- a/go/libraries/doltcore/sqle/dtables/diff_table.go
+++ b/go/libraries/doltcore/sqle/dtables/diff_table.go
@@ -637,7 +637,7 @@ func (dp DiffPartition) Key() []byte {
 
 func (dp DiffPartition) GetRowIter(ctx *sql.Context, ddb *doltdb.DoltDB, joiner *rowconv.Joiner, lookup sql.IndexLookup) (sql.RowIter, error) {
 	if types.IsFormat_DOLT(ddb.Format()) {
-		return newProllyDiffIter(ctx, dp, ddb, dp.fromSch, dp.toSch)
+		return newProllyDiffIter(ctx, dp, dp.fromSch, dp.toSch)
 	} else {
 		return newNomsDiffIter(ctx, ddb, joiner, dp, lookup)
 	}
@@ -647,6 +647,13 @@ func (dp DiffPartition) GetRowIter(ctx *sql.Context, ddb *doltdb.DoltDB, joiner 
 // If the primary key sets changed between the two commits, it may not be
 // possible to diff them.
 func (dp *DiffPartition) isDiffablePartition(ctx *sql.Context) (bool, error) {
+	// dp.to is nil when a table has been deleted previously. In this case, we return
+	// false, to stop processing diffs, since that previously deleted table is considered
+	// a logically different table and we don't want to mix the diffs together.
+	if dp.to == nil {
+		return false, nil
+	}
+
 	// dp.from is nil when the to commit created a new table
 	if dp.from == nil {
 		return true, nil
@@ -656,14 +663,7 @@ func (dp *DiffPartition) isDiffablePartition(ctx *sql.Context) (bool, error) {
 	if err != nil {
 		return false, err
 	}
-
-	// dp.to is nil when a table has been deleted previously. In this case, we return
-	// false, to stop processing diffs, since that previously deleted table is considered
-	// a logically different table and we don't want to mix the diffs together.
-	if dp.to == nil {
-		return false, nil
-	}
-
+	
 	toSch, err := dp.to.GetSchema(ctx)
 	if err != nil {
 		return false, err

--- a/go/libraries/doltcore/sqle/enginetest/dolt_queries_diff.go
+++ b/go/libraries/doltcore/sqle/enginetest/dolt_queries_diff.go
@@ -4777,7 +4777,7 @@ var CommitDiffSystemTableScriptTests = []queries.ScriptTest{
 			"insert into t values (1, 2), (3, 4);",
 			"set @Commit1 = '';",
 			"CALL DOLT_COMMIT_HASH_OUT(@Commit1, '-am', 'creating table t');",
-			
+
 			"drop table t",
 			"set @Commit2 = '';",
 			"CALL DOLT_COMMIT_HASH_OUT(@Commit2, '-am', 'dropping table');",
@@ -4798,14 +4798,14 @@ var CommitDiffSystemTableScriptTests = []queries.ScriptTest{
 				ExpectedWarning:                 1105,
 				ExpectedWarningsCount:           1,
 				ExpectedWarningMessageSubstring: "cannot render full diff between commits",
-				Expected: []sql.Row{},
+				Expected:                        []sql.Row{},
 			},
 			{
 				Query:                           "select * from dolt_commit_diff_t where from_commit=@Commit3 and to_commit=@Commit3;",
 				ExpectedWarning:                 1105,
 				ExpectedWarningsCount:           1,
 				ExpectedWarningMessageSubstring: "cannot render full diff between commits",
-				Expected: []sql.Row{},
+				Expected:                        []sql.Row{},
 			},
 			{
 				Query:    "SELECT to_pk, to_c1, from_pk, from_c1, diff_type FROM DOLT_commit_DIFF_t where from_commit=@Commit3 and to_commit=@Commit4;",

--- a/go/libraries/doltcore/sqle/enginetest/dolt_queries_diff.go
+++ b/go/libraries/doltcore/sqle/enginetest/dolt_queries_diff.go
@@ -4769,6 +4769,54 @@ var CommitDiffSystemTableScriptTests = []queries.ScriptTest{
 			},
 		},
 	},
+	{
+		Name: "added and dropped table",
+		SetUpScript: []string{
+			"create table t (pk int primary key, c1 int);",
+			"call dolt_add('.')",
+			"insert into t values (1, 2), (3, 4);",
+			"set @Commit1 = '';",
+			"CALL DOLT_COMMIT_HASH_OUT(@Commit1, '-am', 'creating table t');",
+			
+			"drop table t",
+			"set @Commit2 = '';",
+			"CALL DOLT_COMMIT_HASH_OUT(@Commit2, '-am', 'dropping table');",
+
+			"create table unrelated (a int primary key);",
+			"set @Commit3 = '';",
+			"CALL DOLT_COMMIT_HASH_OUT(@Commit3, '-Am', 'created unrelated table');",
+
+			"create table t (pk int primary key, c1 int);",
+			"call dolt_add('.')",
+			"insert into t values (1, 2);",
+			"set @Commit4 = '';",
+			"CALL DOLT_COMMIT_HASH_OUT(@Commit4, '-Am', 'recreating table t');",
+		},
+		Assertions: []queries.ScriptTestAssertion{
+			{
+				Query:                           "select * from dolt_commit_diff_t where from_commit=@Commit2 and to_commit=@Commit3;",
+				ExpectedWarning:                 1105,
+				ExpectedWarningsCount:           1,
+				ExpectedWarningMessageSubstring: "cannot render full diff between commits",
+				Expected: []sql.Row{},
+			},
+			{
+				Query:                           "select * from dolt_commit_diff_t where from_commit=@Commit3 and to_commit=@Commit3;",
+				ExpectedWarning:                 1105,
+				ExpectedWarningsCount:           1,
+				ExpectedWarningMessageSubstring: "cannot render full diff between commits",
+				Expected: []sql.Row{},
+			},
+			{
+				Query:    "SELECT to_pk, to_c1, from_pk, from_c1, diff_type FROM DOLT_commit_DIFF_t where from_commit=@Commit3 and to_commit=@Commit4;",
+				Expected: []sql.Row{{1, 2, nil, nil, "added"}},
+			},
+			{
+				Query:    "SELECT to_pk, to_c1, from_pk, from_c1, diff_type FROM DOLT_commit_DIFF_t where from_commit=@Commit1 and to_commit=@Commit4;",
+				Expected: []sql.Row{{nil, nil, 3, 4, "removed"}},
+			},
+		},
+	},
 }
 
 type systabScript struct {

